File: LivrusVirtus/wxWindows/wx.NET/CS/ConEnderecos.cs
using System;
using System.Drawing;
using wx;

namespace Livrus
{
	public class frmConEnderecos : frmConsultas
	{
		public static frmConEnderecos fConEnderecos;

		public frmConEnderecos(wx.Window parent)
      : base(parent){

      Title = Resource.STR_CONENDERECO;
      labDescricao.Label = "Endereço: ";
      gridConsulta.SetColLabelValue(1, "Logradouro");
      gridConsulta.SetColSize(1, 200);
 
      int[] nLargura = new int[4] { 150,80,100,80 };
      string[] sRotulos = new string[4] { "Bairro", 
        "CEP", "Cidade", "Estado" };
      RotinasGlobais.Rotinas.AdicionaColunasGrid(gridConsulta, 2, 4, 
        nLargura, sRotulos);

      EVT_BUTTON((int)idt.ID_FRMCONSULTAS_BTNPESQUISAR, 
        new EventListener( OnBtnPesquisar ));
      EVT_BUTTON((int)idt.ID_FRMCONSULTAS_BTNLIMPAR, 
        new EventListener( OnBtnLimpar ));
      EVT_BUTTON((int)idt.ID_FRMCONSULTAS_BTNFECHAR, 
        new EventListener( OnBtnFechar ));
      EVT_CLOSE(new EventListener( OnClose ));
      EVT_GRID_SELECT_CELL(new EventListener( OnGridSelectCell ));
      EVT_GRID_LABEL_LEFT_CLICK(new EventListener( OnGridLabelLeftClick ));
		}

    private void OnClose( object sender, Event e ){
      Fecha(Resource.MNU_CON_ENDERECOS);
      e.Skip();
    }
    
    private void OnBtnFechar( object sender, Event e ){
      Fecha(0);
      e.Skip();
    }
    
    private void OnBtnLimpar( object sender, Event e ){
      LimparDados();
      e.Skip();
    }
    
    private void OnBtnPesquisar( object sender, Event e ){
      PesquisarDados(ConsultasSQL.ConsSQL.Endereco('S',"", edtDescricao.Value,
        "","","",""), Resource.STR_ENDNENC, 6);
      e.Skip();
    }
    
    private void OnGridSelectCell( object sender, Event e ){
      wx.GridEvent evt = (wx.GridEvent)e;
      setLabRegistros(evt.Row);
      e.Skip();
    }
    
    private void OnGridLabelLeftClick( object sender, Event e ){
      wx.GridEvent evt = (wx.GridEvent)e;
      setLabRegistros(evt.Row);
      e.Skip();
    }

	}
}

File: LivrusVirtus/wxWindows/wx.NET/VB.NET/ConEnderecos.vb
Imports System
Imports System.Drawing
Imports wx

Namespace Livrus

	Public Class frmConEnderecos
		Inherits frmConsultas
		
		public shared WithEvents fConEnderecos as frmConEnderecos

    Public Sub New(ByVal parent as Window)
      MyBase.New(parent)

      Title = Resource.STR_CONENDERECO
      labDescricao.Label = "Endereço: "
      gridConsulta.SetColLabelValue(1, "Logradouro")
      gridConsulta.SetColSize(1, 200)
 
      dim nLargura() as integer = { 150,80,100,80 }
      dim sRotulos() as string = { "Bairro", _
        "CEP", "Cidade", "Estado" }
      RotinasGlobais.Rotinas.AdicionaColunasGrid(gridConsulta, 2, 4, _
        nLargura, sRotulos)

      EVT_BUTTON(idt.ID_FRMCONSULTAS_BTNPESQUISAR, _
        new EventListener(AddressOf OnBtnPesquisar ))
      EVT_BUTTON(idt.ID_FRMCONSULTAS_BTNLIMPAR, _
        new EventListener(AddressOf OnBtnLimpar ))
      EVT_BUTTON(idt.ID_FRMCONSULTAS_BTNFECHAR, _
        new EventListener(AddressOf OnBtnFechar ))
      EVT_CLOSE(new EventListener(AddressOf OnClose ))
      EVT_GRID_SELECT_CELL(new EventListener(AddressOf OnGridSelectCell ))
      EVT_GRID_LABEL_LEFT_CLICK(new EventListener( _
        AddressOf OnGridLabelLeftClick ))
    End Sub

		private sub OnClose(ByVal sender As Object, ByVal e As wx.Event)
      Fecha(Resource.MNU_CON_ENDERECOS)
			e.Skip()
		End Sub
    
    private sub OnBtnFechar(ByVal sender As Object, ByVal e As wx.Event)
      Fecha(0)
      e.Skip()
    end sub
    
    private sub OnBtnLimpar(ByVal sender As Object, ByVal e As wx.Event)
      LimparDados()
      e.Skip()
    end sub
    
    private sub OnBtnPesquisar(ByVal sender As Object, ByVal e As wx.Event)
      PesquisarDados(ConsultasSQL.ConsSQL.Endereco("S","", edtDescricao.Value, _
        "","","",""), Resource.STR_ENDNENC, 6)
      e.Skip()
    end sub
    
    private sub OnGridSelectCell(ByVal sender As Object, ByVal e As wx.Event)
      dim evt as wx.GridEvent = CType(e, wx.GridEvent)
      setLabRegistros(evt.Row)
      e.Skip()
    end sub
    
    private sub OnGridLabelLeftClick(ByVal sender As Object, _
      ByVal e As wx.Event)
      dim evt as wx.GridEvent = CType(e, wx.GridEvent)
      setLabRegistros(evt.Row)
      e.Skip()
    end sub

	End Class

End Namespace
